populate.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fx
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.uber.org/dig"
+)
+
+// Populate fills the given struct with values from the DI container when
+// passed to App.Start.
+//
+// 	var target struct {
+// 		Dispatcher *yarpc.Dispatcher
+// 	}
+// 	err := app.Start(ctx, Populate(&target))
+//
+// The target MUST be a pointer to a struct. Only exported fields will be
+// filled. All field tags supported by dig.In are supported on Populate
+// targets.
+func Populate(target interface{}) interface{} {
+	v := reflect.ValueOf(target)
+
+	if t := v.Type(); t.Kind() != reflect.Ptr || t.Elem().Kind() != reflect.Struct {
+		panic(fmt.Sprintf("expected a pointer to a struct, got a %v", t))
+	}
```

Comment:
Rather than panic, what do you think of returning a function like this:

```
func() error {
   return fmt.Sprintf("Populate expected a pointer to a struct, got a %v", t)
}
```